Agents/FenlandTrajectoryAgent/agent/exposure_calculator/exposure_utilities.py

```diff
@@ -0,0 +1,1023 @@
+            result = self.execute_query(conn, check_sql, (table_name,))
+            has_trip_index = bool(result)
+        except Exception as e:
+            self.logger.warning(f"[Trip index] Failed to check for 'Trip index' column: {e}")
+            has_trip_index = False
+
+        try:
+            if has_trip_index:
+                self.logger.info(f"[Trip index] Detected in table: {table_name}")
+                sql_template_name = "get_timeseries_data_with_trip.sql"
+            else:
+                self.logger.info(f"[Trip index] Not found, using fallback SQL.")
+                sql_template_name = "get_timeseries_data.sql"
+
+            query_template = self.load_template(sql_template_name)
+            query = query_template.format(table_name=table_name)
+            rows = self.execute_query(conn, query)
+
+            base_cols = ["time", "column1", "column2", "column3", "column4", "column5", "column6", "column7"]
+            if has_trip_index:
+                base_cols.append("Trip index")
+
+            return pd.DataFrame(rows, columns=base_cols)
+
+        except Exception as primary_exception:
+            self.logger.warning(f"[Fallback] Primary SQL failed for {table_name}: {primary_exception}")
+            
+            # Run fallback version
+            try:
+                fallback_template = self.load_template("get_timeseries_data_fallback.sql")
+                query = fallback_template.format(table_name=table_name)
+                rows = self.execute_query(conn, query, (trajectory_iri,))
+
+                return pd.DataFrame(rows, columns=["time", "LATITUDE", "LONGITUDE"])
+
+            except Exception as fallback_exception:
+                self.logger.error(f"[Fallback] Failed for {table_name}: {fallback_exception}")
+                raise fallback_exception
+
+
+    def fetch_trajectory_data_from_db(self, trajectory_iri: str) -> pd.DataFrame:
+        """
+        Fetch and process trajectory data from database.
+        Handles basic column renaming and time formatting.
+        """
+        self.logger.info("==== fetch_trajectory_data_from_db() called ====")
+        conn = None
+        try:
+            conn = self.connect_to_database(
+                self.TRAJECTORY_DB_HOST,
+                self.DB_PORT,
+                self.DB_USER,
+                self.TRAJECTORY_DB_PASSWORD,
+                self.DB_NAME
+            )
+            table_name = self.get_table_name_for_timeseries(conn, trajectory_iri)
+            df = self.get_timeseries_data(conn, table_name, trajectory_iri)
+        finally:
+            if conn:
+                conn.close()
+
+        df = df.rename(columns={
+            "column1": "SPEED",
+            "column2": "DISTANCE",
+            "column3": "HEIGHT",
+            "column4": "HEADING",
+            "column5": "LATITUDE",
+            "column6": "LONGITUDE",
+            "column7": "POINT"
+        })
```

Comment:
The sequence of this list is not guaranteed to be the same because it is a Map in TimeSeriesClient, this should not be hardcoded